plugin/impl.go

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) 2020, the Drone Plugins project authors.
+// Please see the AUTHORS file for details. All rights reserved.
+// Use of this source code is governed by an Apache 2.0 license that can be
+// found in the LICENSE file.
+
+package plugin
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"path/filepath"
+	"strings"
+
+	"github.com/google/go-github/v32/github"
+	"github.com/urfave/cli/v2"
+	"golang.org/x/oauth2"
+)
+
+// Settings for the plugin.
+type Settings struct {
+	APIKey          string
+	Files           cli.StringSlice
+	FileExists      string
+	Checksum        cli.StringSlice
+	ChecksumFile    string
+	ChecksumFlatten bool
+	Draft           bool
+	Prerelease      bool
+	BaseURL         string
+	UploadURL       string
+	Title           string
+	Note            string
+	Overwrite       bool
+
+	baseURL   *url.URL
+	uploadURL *url.URL
+	uploads   []string
+}
+
+// Validate handles the settings validation of the plugin.
+func (p *Plugin) Validate() error {
+	var err error
+
+	if p.pipeline.Build.Event != "tag" {
+		return fmt.Errorf("github release plugin is only available for tags")
+	}
+
+	if p.settings.APIKey == "" {
+		return fmt.Errorf("no api key provided")
+	}
+
+	if !fileExistsValues[p.settings.FileExists] {
+		return fmt.Errorf("invalid value for file_exists")
+	}
+
+	if !strings.HasSuffix(p.settings.BaseURL, "/") {
+		p.settings.BaseURL = p.settings.BaseURL + "/"
+	}
+	p.settings.baseURL, err = url.Parse(p.settings.BaseURL)
+	if err != nil {
+		return fmt.Errorf("failed to parse base url: %w", err)
+	}
+
+	if !strings.HasSuffix(p.settings.UploadURL, "/") {
+		p.settings.UploadURL = p.settings.UploadURL + "/"
+	}
+	p.settings.uploadURL, err = url.Parse(p.settings.UploadURL)
+	if err != nil {
+		return fmt.Errorf("failed to parse upload url: %w", err)
+	}
+
+	if p.settings.Note != "" {
+		if p.settings.Note, err = readStringOrFile(p.settings.Note); err != nil {
+			return fmt.Errorf("error while reading %s: %w", p.settings.Note, err)
+		}
+	}
+
+	if p.settings.Title != "" {
+		if p.settings.Title, err = readStringOrFile(p.settings.Title); err != nil {
+			return fmt.Errorf("error while reading %s: %w", p.settings.Note, err)
+		}
+	}
+
+	files := p.settings.Files.Value()
+	for _, glob := range files {
+		globed, err := filepath.Glob(glob)
+
+		if err != nil {
+			return fmt.Errorf("failed to glob %s: %w", glob, err)
+		}
+
+		if globed != nil {
+			p.settings.uploads = append(p.settings.uploads, globed...)
+		}
+	}
+
+	if len(files) > 0 && len(p.settings.uploads) < 1 {
```

Comment:
Just check `len(p.settings.uploads) >= 1`? The rest doesn't really matter.

Reply:
You can cut a GitHub release with no files present. In that case uploads would be 0. I think this logic is actually correct.